addons/udes_stock/controllers/stock_picking.py

```diff
@@ -93,3 +93,14 @@ def batch_to_user(self, ident):
             res = picking.batch_id.get_info(None)[0]
 
         return res
+
+    @http.route('/api/stock-picking/<ident>/warn_picking_precondition',
+                type='json', methods=['GET'], auth='user')
+    def warn_picking_precondition(self, ident):
+        """ Check if picking does not fulfill a precondition.
+            Returns a warning string if it doesn't, False otherwise.
+        """
+        Picking = request.env['stock.picking']
```

Comment:
Nit: Should have a space after `Picking`.

Reply:
Added blank line